src/target/ext/host_io.rs

```diff
@@ -324,36 +320,32 @@ define_ext!(HostIoPwriteOps, HostIoPwrite);
 pub trait HostIoFstat: HostIo {
     /// Get information about the open file corresponding to fd.
     ///
-    /// On success the information is returned as a binary attachment and the
-    /// return value is the size of this attachment in bytes. If an error occurs
-    /// the return value is -1.
-    fn fstat(&mut self, fd: i32) -> HostIoResult<HostStat, Self>;
+    /// On success return a [`HostIoStat`] struct.
+    /// Return [`HostIoError::Errno`] if an error occurs.
+    fn fstat(&mut self, fd: u32) -> HostIoResult<HostIoStat, Self>;
 }
 
 define_ext!(HostIoFstatOps, HostIoFstat);
 
 /// Nested Target Extension - Host I/O unlink operation.
 pub trait HostIoUnlink: HostIo {
-    /// Delete the file at filename on the target.
-    ///
-    /// Return 0, or -1 if an error occurs.
+    /// Delete the file at filename on the target and return 0, or
+    /// [`HostIoError::Errno`] if an error occurs.
     fn unlink(&mut self, filename: &[u8]) -> HostIoResult<u32, Self>;
 }
 
 define_ext!(HostIoUnlinkOps, HostIoUnlink);
 
 /// Nested Target Extension - Host I/O readlink operation.
 pub trait HostIoReadlink: HostIo {
-    /// Read value of symbolic link filename on the target. Return the number of
-    /// bytes read, or -1 if an error occurs.
-    ///
-    /// The data read should be returned as a binary attachment on success. If
-    /// zero bytes were read, the response should include an empty binary
-    /// attachment (i.e. a trailing semicolon).
+    /// Read value of symbolic link filename on the target.
```

Comment:
You went a bit overboard with trimming the docs: you still need to include the line that says "Return the number of bytes read", or else the implementer has no idea what the u32 return value should be (assuming they don't guess) 😅 

Basically, just take a look through all the function docs, and make sure that if you're someone who's new to the vFile spec, you can implement each method without having to crack open the actual verbose GDB RSP docs.

Reply:
Oh, the return type should be `HostIoToken`.

Reply:
ah, yes, exactly.